### DISCUSSION TITLE
Unknown User-Agent that cannot be detected as Android or iOS

### DISCUSSION THREAD

**Discussion by RRGT19** [ionic-team/capacitor#6772] [General] [answered]
Hi community,

```
User-Agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko)
Referer=null
Origin=capacitor://localhost
```

We have developed a mobile app using Ionic and Capacitor. Works awesome and it's in production already. Our backend members have been developer a way to detect if the requests is being made from a mobile app or a browser. They categorize in the case of mobile, if it's Android or iOS by checking the User-Agent HTTP header.

We have noticed a specific User-Agent that cannot be detected as Android/iOS. All we have in this case is the Origin that says `capacitor://localhost`. So far after some investigation, we cannot tell which device it's or why the User-Agent has missing pieces.

I know that some User-Agent are reused across different platforms and browsers that cannot tell more information about the device. **Maybe this is one case?**

Any push on this topic would be appreciated. Thanks a lot.

**Comment by jcesarmobile** [comment, marked as answer]
iPads by default have a desktop user agent instead of a mobile one
You can force them to use a mobile user agent by using `preferredContentMode` config option in capacitor config file like this:

```
"ios": {
        "preferredContentMode": "mobile"
 }
 ```

You could also append your app name to the user agent so it's easier for the backend members to detect it's the mobile app by using `appendUserAgent` like this: 
```
"ios": {
   "appendUserAgent": "yourAppName"
}
```
can also be used inside `android` object to append it to Android, or in the root so it's appended to both iOS and Android.

**Reply by RRGT19** [reply]
Didn't know that, thanks a lot!

From iPad with defaults:
`User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)`

From iPad using `"preferredContentMode": "mobile"`:
`User-Agent: Mozilla/5.0 (iPad; CPU OS 16_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148`

Now, they are recognized correctly.

Thanks again for helping and spending time in your explanation.

**Reply by RRGT19** [reply]
The `appendUserAgent` option would really help us a lot, didn't know that option either. I will coordinate with my team the text to add on both platforms. Our application uses the same name and package id/name, so we will have to add another thing, different on both platforms.

**Reply by jcesarmobile** [reply]
You can append whatever you want, and as you can use it inside ios or android object or on the root, if you want different values per platform just add the `appendUserAgent` in each platform object with different values, the root one is for when you want the same vale on both platforms.
